SimpleTimer.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections;
+using SubsurfaceStudios.Attributes;
+using SubsurfaceStudios.Utilities.Async;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace SubsurfaceStudios.Utilities.Time
+{
+    [System.Serializable]
+    public class SimpleTimer
+    {
+        public float duration;
+        
+        [Header("Runtime")]
+        [InspectorReadOnly] public bool isRunning = false;
+        [InspectorReadOnly] public float currentTime = 0f;
+        
+        public UnityEvent<float> onTimerStart;
+        public UnityEvent<float> onTimerTick;
+        public UnityEvent<float> onTimerStop;
+
+        public SimpleTimer(float duration)
+        {
+            this.duration = duration;
+        }
+        
+        public void Start()
+        {
+            if (isRunning)
+            {
+                Debug.LogError("Timer already running!");
```

Comment:
I would consider making this an exception, although I could also see why it would be better as an error. Up to you!